### ISSUE TITLE
Support Importing Out of Band Applications

### ISSUE THREAD

**Issue by athiththan11** [wso2/product-apim-tooling#718]
**Description:**

The API Controller (v3.2.1) supports exporting Applications that are configured as Out-of-Band in the API Manager environment. But, the tool doesn't support importing the same applications to a newer environment without generating the Keys from the scratch. The controller needs to support importing the Application in the newer environment and then mapping the existing Keys with that specific Application.

**Affected Product Version:**

- API Manager v3.2.0 and newer

**Steps to reproduce:**

- Configure API Manager v3.2.0 with external Key Manager (example: Keycloak)
- Disable the OAuth App Creation for the Keycloak Key Manager in the Admin portal
- Create a client in the Keycloak and then create an Application in the Devportal and provision the Keys
- Once that is done, set up the API Controller, and export the application with the Keys (the exported Zip will contain the Consumer credentials)
- Import the Application using the API Controller to an identical newer environment (which is also configured with Keycloak KM and OAuth App Creation disabled)

This fails with an error trying to generate the Keys (trying to create a client at the Keycloak) when importing the Application

**Potential Fix**

Introduce a new flag (example: `--outOfBand` to specify whether the Application is an out-of-band Application or not and based on the presence, import the Application and map the consumer credentials with the Application without generating the Client (SP).

**Related Issues:**

https://github.com/wso2/product-apim/issues/11125

**Comment by praminda** [comment]
@athiththan11 Can we close this as a duplicate of wso2/product-apim#11125?
Fix for 11125 is done to map the keys instead of creating the app when KM oauth app creation is disabled

**Comment by praminda** [comment]
Therefore i'm closing this issue. Please reopen if this requires another look

**Comment by wasuradananjith** [comment]
Fixed in master via https://github.com/wso2/carbon-apimgt/pull/10479
